pkg/controller/sidecarset/sidecarset_utils.go

```diff
@@ -101,3 +112,182 @@ func inconsistentStatus(sidecarSet *appsv1alpha1.SidecarSet, status *appsv1alpha
 		status.UpdatedPods != sidecarSet.Status.UpdatedPods ||
 		status.ReadyPods != sidecarSet.Status.ReadyPods
 }
+
+// add this cache to avoid be influenced by informer cache latency when controller try to count maxUnavailable
+type updatedPodCache struct {
+	lock sync.RWMutex
+	// key is in the format: sidecarset name/pod namespace/pod name
+	// value is sidecarset hash
+	podSidecarUpdated map[string]string
+}
+
+func (u *updatedPodCache) set(key, hash string) {
+	u.lock.Lock()
+	u.podSidecarUpdated[key] = hash
+	u.lock.Unlock()
+}
+
+func (u *updatedPodCache) reset() {
```

Comment:
do we need a delete method if a pod is deleted, otherwise the cache will grow indefinitely

Reply:
done, add a removeStaleCache function